src/Constraints/ArrayUniqueItemsConstraint.php

```diff
@@ -0,0 +1,27 @@
+<?hh // strict
+
+namespace Slack\Hack\JsonSchema\Constraints;
+
+use namespace HH\Lib\C;
+use namespace Slack\Hack\JsonSchema;
+
+class ArrayUniqueItemsConstraint {
+  public static function check<T as arraykey>(
+    vec<T> $input,
+    string $pointer,
+  ): keyset<T> {
+    $output = keyset($input);
```

Comment:
One thought is that we could check for `coerce` here and if so just return `keyset($input)`. I can't think of a lot of cases where we want to ensure that unique items are provided and reject it vs accepting an array but having keyset called already on it.

Reply:
Hmm, I guess my only concern there is we're overloading `coerce` from its usual meaning. But I agree that deduping items is a more common use-case than rejecting input which contains duplicates, so it'd be good to support somehow, and I can't think of a better way currently. So yeah, I'm inclined to make this change.

Reply:
@mwildehahn updated in https://github.com/slackhq/hack-json-schema/pull/43/commits/edf693cf16ac75adfe52eb86cbc2afd4d27519df